Added Backward Compatibility promise section (#34)
apigee/apigee-client-php: README.md

PS.: Some unit tests cannot be executed when the offline test data set is in use, those are automatically marked as
skipped.

## Our Backward Compatibility (BC) Promise

> This library follows the [Semantic Versioning](https://semver.org) strategy. It means only major releases (such as 3.0, 4.0 etc.) are allowed to break backward compatibility and we do our best to keep it this way.

Reviewer: ```suggestion
This library follows the [Semantic Versioning](https://semver.org) strategy. This means that only major releases (such as 3.0, 4.0 etc.) are allowed to break backward compatibility and we do our best to follow those rules.
```

> The PHP API Client has to be compatible with the latest versions of the supported Apigee Edge APIs.

Reviewer: You said "has to be compatible", but I think you meant to declare that we will support the latest version?  See if this is correct:
```suggestion
The PHP API Client will always be compatible with the latest versions of the supported Apigee Edge APIs.
```

> As you can see, almost every class implements at least one interface in this library, therefore, we would like to suggest to rely on interface definitions instead of concrete classes if you build something on the top of this library. Also, prefer encapsulation over inheritance if you extend our classes.

Reviewer: ```suggestion
As you can see, almost every class implements at least one interface in this library. Therefore, we would like to suggest that you rely on interface definitions instead of concrete classes if you build something on the top of this library. Also, prefer encapsulation over inheritance if you extend our classes.
```
